src/coredump/coredumpctl.c

```diff
@@ -467,6 +516,20 @@ static void analyze_coredump_file(
         *ret_size = UINT64_MAX;
 }
 
+static int resolve_filename(const char *prefix, char **filename) {
```

Comment:
Instead of always modifying the input argument, we can make this more flexible by using separate input/output arguments, something like the following:

```
static int resolve_filename(const char *path, const char *root, char **ret)
```

That way, the input argument can be separate from the output argument. You can still call it as follows though:

```
r = resolve_filename(filename, arg_root, &filename);
```
```